frontend/service/agents/PettAi.ts

```diff
@@ -0,0 +1,253 @@
+import { ethers } from 'ethers';
+import { formatEther } from 'ethers/lib/utils';
+
+import { STAKING_PROGRAMS } from '@/config/stakingPrograms';
+import { BASE_STAKING_PROGRAMS } from '@/config/stakingPrograms/base';
+import { PROVIDERS } from '@/constants/providers';
+import { EvmChainId } from '@/enums/Chain';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { Address } from '@/types/Address';
+import {
+  ServiceStakingDetails,
+  StakingContractDetails,
+  StakingRewardsInfo,
+} from '@/types/Autonolas';
+
+import {
+  ONE_YEAR,
+  StakedAgentService,
+} from './shared-services/StakedAgentService';
+
+const REQUESTS_SAFETY_MARGIN = 1;
+
+export abstract class PettAiService extends StakedAgentService {
+  static getAgentStakingRewardsInfo = async ({
+    agentMultisigAddress,
+    serviceId,
+    stakingProgramId,
+    chainId = EvmChainId.Base,
+  }: {
+    agentMultisigAddress: Address;
+    serviceId: number;
+    stakingProgramId: StakingProgramId;
+    chainId?: EvmChainId;
+  }): Promise<StakingRewardsInfo | undefined> => {
+    if (!agentMultisigAddress) return;
+    if (!serviceId) return;
+    if (serviceId === -1) return;
+
+    const stakingProgramConfig = STAKING_PROGRAMS[chainId][stakingProgramId];
+
+    if (!stakingProgramConfig) throw new Error('Staking program not found');
+
+    const { activityChecker, contract: stakingTokenProxyContract } =
+      stakingProgramConfig;
+
+    const provider = PROVIDERS[chainId].multicallProvider;
+
+    const contractCalls = [
+      stakingTokenProxyContract.getServiceInfo(serviceId),
+      stakingTokenProxyContract.livenessPeriod(),
+      stakingTokenProxyContract.rewardsPerSecond(),
+      stakingTokenProxyContract.calculateStakingReward(serviceId),
+      stakingTokenProxyContract.minStakingDeposit(),
+      stakingTokenProxyContract.tsCheckpoint(),
+      activityChecker.livenessRatio(),
+      activityChecker.getMultisigNonces(agentMultisigAddress),
```

Comment:
even tho the activity checker is different, those two are there, it should be enough to correctly calculate "earned rewards" logic